core/src/sumeragi/network_topology.rs

```diff
@@ -158,38 +158,31 @@ impl Topology {
             .len()
             .try_into()
             .expect("`usize` should fit into `u64`");
-        if let Some(mut rem) = n.checked_rem(len) {
-            // In case where `n` is larger than `usize` could fit
-            let usize_max = usize::MAX
-                .try_into()
-                .expect("`usize` should fit into `u64`");
-            while rem > usize_max {
-                rem -= usize_max;
-                self.ordered_peers.rotate_left(usize::MAX);
-            }
-            let rem = rem
-                .try_into()
-                .expect("`rem` is smaller or equal then `usize::MAX`");
-            self.ordered_peers.rotate_left(rem);
+        if let Some(rem) = n.checked_rem(len) {
+            let rem = rem.try_into().expect(
+                "`rem` is smaller than `usize::MAX`, because remainder is always smaller than divisor",
+            );
+            self.ordered_peers.as_mut_unchecked().rotate_left(rem);
         }
     }
 
     /// Re-arrange the set of peers after each successful block commit.
     pub fn rotate_set_a(&mut self) {
         let rotate_at = self.min_votes_for_commit();
         if rotate_at > 0 {
-            self.ordered_peers[..rotate_at].rotate_left(1);
+            self.ordered_peers.as_mut_unchecked()[..rotate_at].rotate_left(1);
```

Comment:
> Though std unchecked is used to indicate unsafe methods.

Yes, I've also checked for that... I don't know a better name for *Business Logic Errors*.
But in another comment we decided to remove this method, so no problems.

Reply:
> Though `std` `unchecked` is used to indicate `unsafe` methods.

indeed, it seems that `_unchecked` suffix implies `unsafe` in `std` because there is no `_unchecked` method that is not also `unsafe`. But it is my understanding that `_unchecked` warns you of potential logical errors, whereas `unsafe` warns you of potential UB

Reply:
For logic errors usually `panic!` and friends work fine, but in the case we could only do that after. For example every next operation after `_unchecked()` would check for uniqueness and panic if assertion failed.